src/runtime/internal/cache.ts

```diff
@@ -137,6 +137,18 @@ export function defineCachedFunction<T, ArgsT extends unknown[] = any[]>(
       _resolvePromise.catch((error) => {
         console.error(`[nitro] [cache] SWR handler error.`, error);
         useNitroApp().captureError(error, { event, tags: ["cache"] });
+
+        // SWR revalidation failed, remove existing entry so we do
+        // not continue to return the cached value indefinitely.
+        const promise = useStorage()
+          .removeItem(cacheKey)
```

Comment:
we do not store invalid (error, empty) values in cache -- they can be expired though. if SWR handler fails, old value should be served to requests until it can successfully be revalidated.

I'm reverting this part to unblock PR, we can discuss later if you think we still should.